swap/src/protocol/bob/swap.rs

```diff
@@ -190,6 +190,11 @@ async fn next_state(
                 // Bob sends Alice his key
 
                 select! {
+                    state5 = state.watch_for_redeem_btc(bitcoin_wallet) => {
+                        // there seems to be a weird edge case where you can send the encrypted signature, alice can redeem the btc, but your database doesn't update to encsigsent or btcredeemed, so here we will watch for redemption too, so we have a chance to unlock xmr.
+                        // this seemed to have happened when a surge of blocks came in on stagenet within a short time frame
```

Comment:
Perhaps replace this comment with something simpler:

"If we miss the enc sign acknowledgement sent by the ASB for some reason (client crash, network issues, ...) the ASB won't send it again.
To avoid staying stuck in XmrLocked we also watch for the btc redeem transaction to then jump to BtcRedeemed"